include/net/socket.h

```diff
@@ -43,6 +43,16 @@ struct zsock_pollfd {
 #define ZSOCK_MSG_PEEK 0x02
 #define ZSOCK_MSG_DONTWAIT 0x40
 
+/* socket option for TCP */
+#define SOL_TCP             6
+
+/*  Socket option for TLS */
+#define SOL_TLS             282
+
+/* TLS socket options */
+#define TLS_ENABLE          1
+#define TLS_SEC_TAG_LIST    2
```

Comment:
I'm surprised that with all the fuzz about setsockopt, these 2 options is all we get. Why use this 2-storey design where there's TLS_SEC_TAG_LIST and then other options hanging off from it, instead of just adding individual TLS options as socket options?


@GAnthony: Have you seen this? Is this what you expected?

Reply:
Actually, as TLS can be enabled during socket creation, `TLS_ENABLE` is not even needed (I don't think it's a good idea to repurpose the socket after it's created, just like we don't switch between UDP/TCP). 

`TLS_SEC_TAG_LIST` is all about credentials configuration, not more. And in basic TLS use case, it's enough. Once TLS is enabled, all you need to set is credentials to use.

Of course there is a field for adding more options (if needed of course, I don't see a releation between more options - better solution, as you indicate). For instance DTLS will likely need some role configuration.

Reply:
> Actually, as TLS can be enabled during socket creation, TLS_ENABLE is not even needed (I don't think it's a good idea

It's not only good idea, it's how TLS works (it's a separate layer on top of sockets), and is required for the general TLS operation.

> TLS_SEC_TAG_LIST is all about credentials configuration, not more.

Repeating the question from the other comment - how other TLS param are configured? It would be nice to add at least one of such params for completeness, even for the initial PR.